### PR TITLE
fix(dev): add error messages for upload URL expiry

### PR DESCRIPTION
**Pull request by chengoramazon**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Problem

- Upload url would expire in a given time. The original error message is not clear for users what to do next.

### Solution

- Handle URL expiry explicitly.
- Add a clear error message.

JBExpired

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 1]
```diff
@@ -72,6 +72,7 @@ amazonqFeatureDev.exception.request_failed=Request failed
 amazonqFeatureDev.exception.retry_request_failed=Retry request failed
 amazonqFeatureDev.exception.throttling=I'm sorry, I'm experiencing high demand at the moment and can't generate your code. This attempt won't count toward usage limits. Please try again.
 amazonqFeatureDev.exception.upload_code=I'm sorry, I couldn't upload your workspace artifacts to Amazon S3 to help you with this task. You might need to allow access to the S3 bucket. For more information, see the [Amazon Q documentation](https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/security_iam_manage-access-with-policies.html#data-perimeters) or contact your network or organization administrator.
+amazonqFeatureDev.exception.upload_url_expiry=I'm sorry, I wasn't able to generate code. A connection timed out or became unavailable. Please try again or check the following:\n\n- Exclude non-essential files in your workspace?s `.gitignore.`\n\n- Check that your network connection is stable.
```
Comment (willyyhuang) [inline plugins/core/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:75]: typo here 

> workspace?s

File: .changes/next-release/bugfix-caf370b9-c943-4ece-a4c0-81bcefa7a4f5.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Q dev add error messages for upload URL expiry"
```
Comment (willyyhuang) [inline .changes/next-release/bugfix-caf370b9-c943-4ece-a4c0-81bcefa7a4f5.json:3]: probably should make this description a little easier to understand or more vague

> Amazon Q feature dev: Add error message for connection timeout

Comment (chengoramazon) [reply .changes/next-release/bugfix-caf370b9-c943-4ece-a4c0-81bcefa7a4f5.json:3]: It is for upload URL explicitly

**Review by willyyhuang — APPROVED** [review]

File: .changes/next-release/bugfix-caf370b9-c943-4ece-a4c0-81bcefa7a4f5.json (hunk 2 above)
Comment (rli) [inline .changes/next-release/bugfix-caf370b9-c943-4ece-a4c0-81bcefa7a4f5.json:3]: Please put more effort into these descriptions
```suggestion
  "description" : "Amazon Q Feature Dev: Add error messages when the upload URL expires"
```

**Review by rli — APPROVED** [review]

**Pull request merged**